blueflood-core/src/main/java/com/rackspacecloud/blueflood/service/RollupBatchWriter.java

```diff
@@ -53,16 +61,31 @@ public void enqueueRollupForWrite(SingleRollupWriteContext rollupWriteContext) {
     }
 
     public synchronized void drainBatch() {
-        ArrayList<SingleRollupWriteContext> writeContexts = new ArrayList<SingleRollupWriteContext>();
+        List<SingleRollupWriteContext> writeBasicContexts = new ArrayList<SingleRollupWriteContext>();
+        List<SingleRollupWriteContext> writePreAggrContexts = new ArrayList<SingleRollupWriteContext>();
         try {
             for (int i=0; i<=ROLLUP_BATCH_MAX_SIZE; i++) {
-                writeContexts.add(rollupQueue.remove());
+                SingleRollupWriteContext context = rollupQueue.remove();
+                if ( context.getRollup().getRollupType() == RollupType.BF_BASIC ) {
+                    writeBasicContexts.add(context);
+                } else {
+                    writePreAggrContexts.add(context);
+                }
```

Comment:
I think moving this functionality to MetricsRWDelegator will muddy the water. I would have to move the ThreadExecutor stuff to MetricsRWDelegator. I'd rather like to keep the MetricsRWDelegator simpler.